File: Gifs_Watcher/app/src/main/java/com/example/gifs_watcher/managers/CacheManager.java
package com.example.phonenetworks.managers;

import com.example.phonenetworks.models.Record;

import java.util.ArrayList;

public class CacheManager {

    private static CacheManager instance;

    private ArrayList<Record> datas = null;

    public static CacheManager getInstance() {
        if (instance == null) {
            instance = new CacheManager();
        }

        return instance;
    }

    private CacheManager() {
        this.datas = new ArrayList<>();
    };

    public void setArray(ArrayList<Record> datas) {
        this.datas = datas;
    }

    public void addArray(ArrayList<Record> datas) {
        this.datas.addAll(datas);
    }

    public ArrayList<Record> getDatas() { return this.datas; }
}

File: Gifs_Watcher/app/src/main/java/com/example/gifs_watcher/managers/CacheManager.kt
package com.example.gifs_watcher.managers

import com.example.gifs_watcher.models.Results

object CacheManager {

    private var data : ArrayList<Results> = arrayListOf()

    fun setArray(tenorData : ArrayList<Results>) : Unit {
        this.data = tenorData
    }
    fun addArray(tenorData : ArrayList<Results>) : Unit {
        this.data.addAll(tenorData)
    }
    fun getData() : ArrayList<Results> {
        return this.data
    }
}